tests/mail/tests.py

```diff
@@ -46,6 +46,16 @@ def assertMessageHasHeaders(self, message, headers):
                         'the following headers: %s' % (headers - msg_headers),)
 
 
+class FakeSMTPConnection(SMTP):
+    """
+    A SMTP connection that doesn't call any network services on initialization.
+    Helps to simulate an already open connection.
+    """
+
+    def __init__(self, host='', port=0, local_hostname=None, **kwargs):
```

Comment:
Why do you accept argument here, if you don't pass them to super.
Maybe you can just the the correct default for `local_hostname` here and pass the variable.

```suggestion
    def __init__(self, host='', port=0, local_hostname='[127.0.0.1], **kwargs):
        super()__init__(host=host, port=port, local_hostname=local_hostname, **kwargs)
```

Reply:
I don't think that fakeness of the object should depend on args provided to it. If we accept your suggestion then someone can call that fake class with random host arg and think that it still fake while it not anymore.

Maybe poping from kwargs will look less confusing?

```
def __init__(self, **kwargs):
    kwargs.pop('host')
    kwargs.pop('localhost_name')
    super().__init__(host='', local_hostname='[127.0.0.1], **kwargs)
```

Reply:
It looks strange and sounds strange, but I see your point. Yes, it's fake and should be fake even with real input. Maybe we can actually leave it as is, but with a comment?